orb-ui/cone/src/lib.rs

```diff
@@ -0,0 +1,159 @@
+pub mod button;
+pub mod lcd;
+pub mod led;
+
+use crate::button::Button;
+use crate::lcd::{Lcd, LcdCommand};
+use crate::led::{Led, CONE_LED_COUNT};
+use color_eyre::eyre;
+use color_eyre::eyre::Context;
+use embedded_graphics::pixelcolor::{Rgb565, RgbColor};
+use ftdi_embedded_hal::libftd2xx::{Ft4232h, Ftdi, FtdiCommon};
+use image::{ImageFormat, Luma};
+use orb_rgb::Argb;
+use std::sync::{Arc, Mutex};
+use std::{env, fs};
+use tokio::sync::mpsc;
+
+const CONE_FTDI_DEVICE_COUNT: usize = 8;
+
+#[derive(Debug)]
+enum Status {
+    Connected,
+    Disconnected,
+}
+
+/// Cone can be created only if connected to the host over USB.
+pub struct Cone {
+    connection_status: Arc<Mutex<Status>>,
+    lcd: Lcd,
+    led_strip: Arc<Mutex<Led>>,
+    _button: Button,
+}
+
+pub enum ConeEvents {
+    ButtonPressed(bool),
+}
+
+impl Cone {
+    /// Create a new Cone instance.
+    pub fn new(event_queue: mpsc::UnboundedSender<ConeEvents>) -> eyre::Result<Self> {
```

Comment:
Should take all of the JoinHandles of the various tasks, and aggregate them into a struct, and return that as part of a tuple.

Also, lets call it `spawn` instead of `new` so its more clear that we are running a bunch of tasks.

```rust
#[derive(Debug)]
struct ConeJoinHandles {
  pub lcd: LcdJoinHandle,
  pub led: LedJoinHandle,
  pub button: ButtonJoinHandle,
}

impl ConeJoinHandles {
  async fn join(self) -> eyre::Result<()> {
    let _: ((),(),()) = tokio::try_join!(
      // ...
    )?;

    Ok(())
  }
}
```
See [what we did in mcu-util](https://github.com/worldcoin/orb-software/blob/9e2e3908c81de12f95b5a69b8a7fefd3baa0b209/mcu-util/src/orb/mod.rs#L264-L298) for inspiration.